DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -486,10 +507,10 @@ THEN CAST(1 AS BIT)
             );
         }
 
-        public (int? centreId, int? courseCategoryId) GetCourseValidationDetails(int customisationId)
+        public (int? centreId, int? courseCategoryId, bool? allCentres) GetCourseValidationDetails(int customisationId)
         {
-            return connection.QueryFirstOrDefault<(int?, int?)>(
-                @"SELECT c.CentreId, a.CourseCategoryId
+            return connection.QueryFirstOrDefault<(int?, int?, bool?)>(
+                @"SELECT c.CentreId, a.CourseCategoryId, c.AllCentres
                         FROM Customisations AS c
                         INNER JOIN Applications AS a on a.ApplicationID = c.ApplicationID
                         WHERE CustomisationID = @customisationId",

```

Comment:
I've refactored this a fair bit now, changing the pair of service methods to validate against the existence of CentreApplications records as well. I think all the logic is now correct, but it could do with a look over.

Reply:
I had a look over the logic and I think its all fine. Just left a tiny comment on the SQL.